src/setup-python.ts

```diff
@@ -75,35 +75,43 @@ async function run() {
     `Python is expected to be installed into ${process.env['RUNNER_TOOL_CACHE']}`
   );
   try {
-    const version = resolveVersionInput();
+    let versions: string[];
+    const resolvedVersionInput = resolveVersionInput();
     const checkLatest = core.getBooleanInput('check-latest');
 
-    if (version) {
-      let pythonVersion: string;
-      const arch: string = core.getInput('architecture') || os.arch();
-      const updateEnvironment = core.getBooleanInput('update-environment');
-      if (isPyPyVersion(version)) {
-        const installed = await finderPyPy.findPyPyVersion(
-          version,
-          arch,
-          updateEnvironment,
-          checkLatest
-        );
-        pythonVersion = `${installed.resolvedPyPyVersion}-${installed.resolvedPythonVersion}`;
-        core.info(
-          `Successfully set up PyPy ${installed.resolvedPyPyVersion} with Python (${installed.resolvedPythonVersion})`
-        );
-      } else {
-        const installed = await finder.useCpythonVersion(
-          version,
-          arch,
-          updateEnvironment,
-          checkLatest
-        );
-        pythonVersion = installed.version;
-        core.info(`Successfully set up ${installed.impl} (${pythonVersion})`);
-      }
+    if (Array.isArray(resolvedVersionInput)) {
+      versions = resolvedVersionInput as string[];
+    } else {
+      versions = [resolvedVersionInput as string];
+    }
 
+    if (versions.length) {
+      let pythonVersion = '';
+      for (const version of versions) {
+        const arch: string = core.getInput('architecture') || os.arch();
+        const updateEnvironment = core.getBooleanInput('update-environment');
+        if (isPyPyVersion(version)) {
```

Comment:
I think we can get the `arch` and `updateEnvironment` inputs once before the loop. Also we can move getting the `checkLatest` input into the `if` statement, since it is not needed if the version is not found.
```suggestion
      let pythonVersion = '';
      const arch: string = core.getInput('architecture') || os.arch();
      const updateEnvironment = core.getBooleanInput('update-environment');
      
      for (const version of versions) {
        if (isPyPyVersion(version)) {
```